### ISSUE TITLE
Whitespaced block inconsistencies

### ISSUE THREAD

**Issue by wcjohnson** [wcjohnson/babylon-lightscript#2]
### Input Code
```js
if a: {x} else b
```

```js
y = if a: {x} else b
```

### Current Behavior
```js
if (a) {
  x;
} else b;
```

```
Parsing error: Unexpected token, expected :
```

### Expected Behavior
There should be consistency in whether this is legal syntax or not. 

### Resolution
Mixing {} and single-statement blocks should be legal as it is in JS. Mixing whiteblocks with any other type of blocks should be illegal.

**Comment by rattrayalex** [comment]
I thought we fixed this? Did you forget the `:` in the `else:` bit?

**Comment by wcjohnson** [comment]
As of babylon-lightscript/lightscript:

- `parseIfExpression` only allows whiteblock or brace syntax, but it allows them to be mix and matched. Single-line statements are flatly forbidden

- `parseIfStatement` allows whiteblock, braces, or single-line statements ala JS, and prevents the one special case of `else`, followed by a newline, followed by a single-line statement. (Why we settled on that one special case I don't remember but, clearly that is not correct as the above example shows.)

- These two syntaxes are inconsistent with each other, so a programmer can write the same exact thing and just suddenly have it be illegal or legal if he switches it from an expr to a statement or vice versa.

As of my proposed fix:

- Pick either whiteblock or traditional JS syntax.

- Traditional JS syntax works just as it did in JS, so all JS if syntax is still legal as it should be.

- If you pick whiteblock syntax it must be used consistently on both sides of the if, inconsistent use is a parse error.

- This behavior is now consistent across both exprs and statements, so nobody will step on a land mine when they convert one to the other or vice versa.

**Comment by rattrayalex** [comment]
To be clear, this boils down to `if cond: ... else ...` being illegal, right? 

If so, I'm open to a PR for that

**Comment by wcjohnson** [comment]
@rattrayalex 

Not quite. The fix implemented in 0.4.9 on my end is:

- Parser: https://github.com/wcjohnson/babylon-lightscript/commit/90646e6ac5a550a21c2e1fd8ac778902b95ef872

- Compiler: https://github.com/wcjohnson/babel-plugin-lightscript/commit/075779de37fd99f8a1c1a9bebec1d8848252f48f

This gives full consistency between syntax for if statements and if expressions.

**Comment by wcjohnson** [comment]
Here's a rough summary from the docs:

### Deviations from LightScript proper

#### If Statements and Expressions

(See https://github.com/wcjohnson/babylon-lightscript/issues/2)

`@oigroup/babylon-lightscript` enforces consistent syntax between if statements and if expressions, so that programmers don't encounter land mines when switching between the two. In practical terms, this means:

**This is now illegal syntax:**
```js
if a: {x} else b
```
If you use colon syntax on one clause of an `if` you must use it on all clauses.

**This is now legal syntax:**
```js
y = if(three()) 3 else 4
```
Traditional JavaScript `if` syntax now works with `if` expressions (as it should, since it works for `if` statements).
